montepython/io_mp.py

```diff
@@ -284,11 +284,22 @@ def create_output_files(command_line, data):
         data.out_name = os.path.join(
             command_line.folder, outname_base)+str(suffix)+'.txt'
         print('Creating %s\n' % data.out_name)
+        if command_line.print_cosmo:
+            data.cosmo_out_name = os.path.join(
+                command_line.folder, outname_base)+str(suffix)+'.cosmo'
+            data.cosmo_out = open(data.cosmo_out_name, 'w')
+            print('Creating %s\n' % data.cosmo_out_name)
     else:
         data.out_name = os.path.join(
             command_line.folder, outname_base)+command_line.chain_number+'.txt'
         data.out = open(data.out_name, 'w')
         print('Creating %s\n' % data.out_name)
+        if command_line.print_cosmo:
+            data.cosmo_out_name = os.path.join(
+                command_line.folder, outname_base)+command_line.chain_number+'.cosmo'
+            data.cosmo_out = open(data.cosmo_out_name, 'w')
```

Comment:
I don't think it is particularly nice to leave files open like this, but I acknowledge that this is how MontePython is coded. (This is why the last line of chain files are sometimes corrupted!)
In a perfect world, you always open files using a context manager (i.e. using `with`)

Reply:
I agree! But I think, for readability, it's better to keep it like how the usual .txt files are written.